### DISCUSSION TITLE
Functions do not see types from default module

### DISCUSSION THREAD

**Discussion by Fogapod** [geldata/gel#1588] [Q&A] [answered]
This is my code:
```edgeql
START TRANSACTION;
START MIGRATION TO {
    module default {
        type User {
            required property nickname -> str;
            required property email -> str;
            required property email_verified -> bool {
                default := false;
            };
            required property password -> bytes;
            required property created_at -> datetime {
                default := datetime_current();
            };
        }

        function create_user(nickname: str, ema: str, pass: bytes) -> optional User
        {
            volatility := 'VOLATILE';
            using EdgeQL $$
                WITH
                    MODULE deafult
                    conflicting := (
                        SELECT User {
                            created_at,
                        }
                        FILTER .nickname = <str>$nickname OR .email = <str>$email
                        LIMIT 2
                    )
                SELECT conflicting
            $$;
        };
    };
};
```
It gives an error:  `object type or alias 'User' does not exist`
unless I add `MODULE default` at the beginning of `WITH` block or add `default::` prefix.
Is it intended behaviour or bug? Shouldn't types from `default` module be visible by default?

**Comment by elprans** [comment, marked as answer]
It's a bug. Currently, when an expression in schema is seen, the current module is injected automatically into a `WITH` clause.  However, it seems that it is _appended_ to the `WITH` clause instead of being injected as the first entry, and so your first declaration in the `WITH` block does not "see" it (`WITH` entries are processed in order).

@vpetrovykh is currently reworking the expression normalization in #1522 to inject fully-qualifed names instead of injecting an alias, which would make it more robust.
